wire-ios/Wire-iOS/Sources/UserInterface/Conversation Options/CreatePasswordSecuredLink/CreateSecureGuestLinkViewController.swift

```diff
@@ -105,16 +206,71 @@ class CreateSecureGuestLinkViewController: UIViewController, CreatePasswordSecur
         viewModel.requestRandomPassword()
     }
 
+    // MARK: - Validation
+
+    func displayPasswordErrorState(for textFields: [UITextField], for labels: [UILabel]) {
+        for textField in textFields {
+            textField.textColor = validationErrorTextColor
+            textField.layer.borderColor = validationErrorTextColor.cgColor
+        }
+
+        for label in labels {
+            label.textColor = validationErrorTextColor
+        }
+
+    }
+
+    func resetPasswordDefaultState(for textFields: [UITextField], for labels: [UILabel]) {
+        for textField in textFields {
+            textField.applyStyle(.default)
+        }
+
+        for label in labels {
+            label.textColor = SemanticColors.Label.textFieldFloatingLabel
+        }
+
+    }
+
+    @objc
+    func handlePasswordValidation(for textField: ValidatedTextField) -> Bool {
+        let labels: [UILabel] = textField == securedGuestLinkPasswordTextfield ? [passwordRequirementsLabel, setPasswordLabel] : [confirmPasswordLabel]
+
+        let isValid = viewModel.validatePassword(for: textField, against: securedGuestLinkPasswordTextfield)
```

Comment:
here we check if `securedGuestLinkPasswordTextfield` is valid, but the method applies to `securedGuestLinkPasswordValidatedTextField` too. Should be a different validation for `securedGuestLinkPasswordValidatedTextField` ?

Reply:
It works both ways. There is no need for different validation for securedGuestLinkPasswordValidatedTextField

We do all checks in:

```swift
func validatePassword(
        for textField: ValidatedTextField,
        against confirmPasswordField: ValidatedTextField
    ) -> Bool {

        guard let enteredPassword = textField.text,
              !enteredPassword.isEmpty,
              textField.isValid,
              confirmPasswordField.text == enteredPassword else {
            return false
        }

        return true
    }
```

Reply:
The textField (which is an instance of ValidatedTextField) is set to call the handlePasswordValidation(for:) method on the target self whenever its text changes (due to the .editingChanged event). When this happens, the textField itself (the sender) gets passed as the argument to the handlePasswordValidation(for:) method.

So in this case:

```swift
private lazy var securedGuestLinkPasswordValidatedTextField: ValidatedTextField = {
        let textField = ValidatedTextField(
            kind: .password(.guestLinkPassword, isNew: true),
            leftInset: 8,
            accessoryTrailingInset: 0,
            cornerRadius: 12,
            setNewColors: true,
            style: .default
        )
        textField.showConfirmButton = false
        textField.addTarget(self, action: #selector(handlePasswordValidation(for:)), for: .editingChanged)
        textField.addRevealButton(delegate: self)
        textField.placeholder = SecuredGuestLinkWithPasswordLocale.VerifyPasswordTextField.placeholder
        textField.addDoneButtonOnKeyboard()
        textField.returnKeyType = .done

        return textField
    }()
    ```
    
    `securedGuestLinkPasswordValidatedTextField` is being passed as the argument to the handlePasswordValidation method. 

And in the case of the `securedGuestLinkPasswordTextField`, `securedGuestLinkPasswordTextField` is being passed as the argument to the handlePasswordValidation method.